mk/e2e.new.mk

```diff
@@ -124,32 +122,37 @@ test/e2e/debug: $(E2E_DEPS_TARGETS)
 # test/e2e/debug-universal is the same target as 'test/e2e/debug' but builds only 'kuma-universal' image
 # and doesn't start Kind clusters
 .PHONY: test/e2e/debug-universal
-test/e2e/debug-universal: $(E2E_DEPS_TARGETS) build/kumactl images/test k3d/network/create
+test/e2e/debug-universal: $(E2E_DEPS_TARGETS) $(E2E_UNIVERSAL_BIN_DEPS) k3d/network/create
+	$(MAKE) docker/tag/test
 	$(E2E_ENV_VARS) \
 	GINKGO_EDITOR_INTEGRATION=true \
 		$(GINKGO_TEST_E2E) --keep-going=false --procs 1 --fail-fast $(UNIVERSAL_E2E_PKG_LIST)
 
 
 .PHONY: test/e2e
-test/e2e: $(E2E_DEPS_TARGETS) $(E2E_BIN_DEPS)
+test/e2e: $(E2E_DEPS_TARGETS) $(E2E_K8S_BIN_DEPS)
+	$(MAKE) docker/tag
 	$(MAKE) test/e2e/k8s/start
 	$(E2E_ENV_VARS) $(GINKGO_TEST_E2E) --procs 1 $(E2E_PKG_LIST) || (ret=$$?; $(MAKE) test/e2e/k8s/stop && exit $$ret)
 	$(MAKE) test/e2e/k8s/stop
 
 .PHONY: test/e2e-kubernetes
-test/e2e-kubernetes: $(E2E_DEPS_TARGETS) $(E2E_BIN_DEPS)
+test/e2e-kubernetes: $(E2E_DEPS_TARGETS) $(E2E_K8S_BIN_DEPS)
+	$(MAKE) docker/tag
```

Comment:
We only need to tag for running e2e tests.
The reason is that before we would build images without the arch in the tag (implicitly using the arch from the building host).
When releasing we do use these tagged images and then create a manifest. Manifest only exist in docker hub (no way to have them locally).

Therefore when running tests you need to explicitly say which arch you use and tagging seems to be the most appropriate.

Some advantages of this:

1. you can build arches in parallel
2. because all intermediary images are tagged with the arch the risk of crossing wires and getting a binary from the wrong arch is reduced
3. on Darwin or in places with QEMU you can run build and e2e tests for other arches simply by setting: `ENABLED_GOARCHES=amd64`